python/lsst/ts/standardscripts/maintel/calibration/power_on_tunablelaser.py

```diff
@@ -0,0 +1,166 @@
+            title: PowerOnTunableLaser v1
+            description: Configuration for PowerOnTunableLaser.
+              Each attribute can be specified as a scalar or array.
+              All arrays must have the same length (one item per image).
+            type: object
+            properties:
+              sequence_name:
+                description: Name of sequence in MTCalsys
+                type: string
+
+            additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config : ``self.cmd_configure.DataType``
+
+        """
+        self.log.info("Configure started")
+        if self.mtcalsys is None:
+            self.log.debug("Creating MTCalSys.")
+            self.mtcalsys = MTCalsys(
+                domain=self.domain, log=self.log, latiss=self.latiss
+            )
+            await self.mtcalsys.start_task
+
+        self.sequence_name = config.sequence_name
+        self.mtcalsys.load_calibration_config_file()
+        self.mtcalsys.assert_valid_configuration_option(name=self.sequence_name)
+
+        self.config_data = self.mtcalsys.get_calibration_configuration(
+            self.sequence_name
+        )
+        self.laser_mode = self.config_data.laser_mode
+        self.optical_configuration = self.config_data.optical_configuration
+        self.wavelength = self.config_data.wavelength
+
+        if self.laser is None:
+            self.laser = salobj.Remote(
+                domain=self.domain,
+                name="TunableLaser",
+            )
+
+        self.laser.start_task
+
+        self.log.info("Configure completed")
+
+    async def run(self):
+        """Run script."""
+        await self.assert_components_enabled()
+
+        await self.checkpoint("Configuring TunableLaser")
+        await self.configure_tunablelaser()
+
+        await self.checkpoint("Starting laser propagation")
+        await self.start_propagation_on()
+
+    async def start_propagation_on(self):
+        """Starts propagation of the laser"""
+
+        await MTCalsys.laser_start_propagate.start()
+
+    async def configure_tunablelaser(self):
+        """Configure the TunableLaser for the mode and optical configuration"""
+        await MTCalsys.setup_laser(
+            mode=self.laser_mode,
+            wavelength=self.wavelength,
+            optical_configuration=self.optical_configuration,
+            use_projector=False,
+        )
+
+        params = await self.get_tunablelaser_parameters()
+
+        self.log.info(
+            f"Laser Configuration is {params[0]}, \n"
+            f"wavelength is {params[1]}, \n"
+            f"Interlock is {params[2]}, \n"
+            f"Burst mode is {params[3]}, \n"
+            f"Cont. mode is {params[4]}"
+        )
+
+    async def get_tunablelaser_parameters(self):
```

Comment:
While we're adding methods, this seems like a super useful method to add to the MTCalSys class since I suspect it will get called in most tunable laser scripts.

Reply:
Good idea